### PR TITLE
364 spa support merged examples update

### LINKED ISSUE #664: update examples & tutorials to work with POST logout and json responses

### PR DESCRIPTION
**Pull request by dogeared**
Resolves #664

### CODE REVIEW

File: examples/spring-security-spring-boot-webmvc-bare-bones/pom.xml [hunk 1]
```diff
@@ -51,12 +45,6 @@
 
         <!-- Runtime-only dependencies: -->
         <dependency>
-            <groupId>com.stormpath.sdk</groupId>
-            <artifactId>stormpath-sdk-httpclient</artifactId>
-            <version>${project.version}</version>
-            <scope>runtime</scope>
-        </dependency>
-        <dependency>
```
Comment (mraible) [inline examples/spring-security-spring-boot-webmvc-bare-bones/pom.xml]: Why did you remove these two dependencies? Are they included by a parent project?

File: examples/spring-security-spring-boot-webmvc-bare-bones/src/main/resources/static/index.html [hunk 2]
```diff
@@ -1 +1,20 @@
...
+        <link href="https://netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css" rel="stylesheet" type="text/css"/>
+    </head>
+    <body>
+        <div class="container">
+            <h1>Congrats! You Made it!</h1>
+            <form method="POST" action="/logout">
+                <input type="submit" class="btn btn-danger" value="Logout"/>
+            </form>
+        </div>
+        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
```
Comment (mraible) [inline examples/spring-security-spring-boot-webmvc-bare-bones/src/main/resources/static/index.html]: Why "https" for Bootstrap, but "//" for jQuery?

File: examples/spring-security-spring-boot-webmvc-bare-bones/src/main/resources/static/index.html [hunk 3]
```diff
@@ -1 +1,20 @@
...
+        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
+        <meta charset="utf-8"/>
+        <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
+        <meta name="viewport" content="width=device-width, initial-scale=1"/>
+        <link href="https://netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css" rel="stylesheet" type="text/css"/>
+    </head>
+    <body>
+        <div class="container">
+            <h1>Congrats! You Made it!</h1>
+            <form method="POST" action="/logout">
```
Comment (mraible) [inline examples/spring-security-spring-boot-webmvc-bare-bones/src/main/resources/static/index.html]: In the other examples, you use `method="post"`. Is consistency important?

File: examples/spring-webmvc/src/main/webapp/WEB-INF/web.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<web-app xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://java.sun.com/xml/ns/javaee"
+         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"
+         version="3.0">
+</web-app>
```
Comment (mraible) [inline examples/spring-webmvc/src/main/webapp/WEB-INF/web.xml]: Should we be using a 3.1 descriptor here?

```xml
<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
         version="3.1">
</web-app>
```

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java [hunk 5]
```diff
@@ -227,7 +227,8 @@ public void init(HttpSecurity http) throws Exception {
 
             http.authorizeRequests()
                 .antMatchers("/assets/css/stormpath.css").permitAll()
-                .antMatchers("/assets/css/custom.stormpath.css").permitAll();
+                .antMatchers("/assets/css/custom.stormpath.css").permitAll()
+                .antMatchers("/assets/js/stormpath.js").permitAll();
```
Comment (mraible) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: Are the following properties in `web.stormpath.properties` used to configure this?

```
stormpath.web.uris./assets/css/stormpath.css = anon
stormpath.web.uris./assets/css/custom.stormpath.css = anon
stormpath.web.uris./assets/js/stormpath.js=anon
```

**Comment by mraible** [comment]
@dogeared I went through the following examples and started them using `mvn tomcat7:run` or `mvn spring-boot:run`. The POST logout looks good on all of them. The last two have a 404 on `http://localhost:8080/assets/js/stormpath.js`:

```yaml
examples
  - servlet
  - spring-boot-default
  - spring-boot-webmvc
  - spring-security-spring-boot-webmvc
  - spring-security-spring-boot-webmvc-bare-bones
  - spring-security-webmvc
  - spring-webmvc
```

**Comment by mraible** [comment]
Back to you @dogeared.

File: examples/spring-security-spring-boot-webmvc-bare-bones/pom.xml (hunk 1 above)
Comment (dogeared) [inline examples/spring-security-spring-boot-webmvc-bare-bones/pom.xml]: The dependency is not needed as it's included in `stormpath-spring-security-webmvc-spring-boot-starter` already has it.

File: examples/spring-security-spring-boot-webmvc-bare-bones/src/main/resources/static/index.html (hunk 2 above)
Comment (dogeared) [inline examples/spring-security-spring-boot-webmvc-bare-bones/src/main/resources/static/index.html]: Good question. It was that way to begin with and I didn't mess with it. I suppose they could all be `//`

File: examples/spring-webmvc/src/main/webapp/WEB-INF/web.xml (hunk 4 above)
Comment (dogeared) [inline examples/spring-webmvc/src/main/webapp/WEB-INF/web.xml]: Yup. I will update.

File: extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java (hunk 5 above)
Comment (dogeared) [inline extensions/spring/stormpath-spring-security-webmvc/src/main/java/com/stormpath/spring/config/StormpathWebSecurityConfigurer.java]: no. That's only for servlet.

**Comment by dogeared** [comment]
@mraible - I believe I've addressed everything. Back to you!

**Comment by mraible** [comment]
@dogeared I ran all the examples again and tried logging in 1) with invalid credentials, 2) with valid credentials and 3) logged out after logging in successfully. I found several examples do not show an error message with invalid credentials. I'm not sure if this is caused by #661.

PASS
```yaml
examples
  - servlet
  - spring-boot-webmvc
  - spring-webmvc
```

FAIL - no error displayed on invalid credentials, URL is http://localhost:8080/login?error
```yaml
examples
  - spring-boot-default 
  - spring-security-spring-boot-webmvc 
  - spring-security-spring-boot-webmvc-bare-bones 
  - spring-security-webmvc 
```

**Comment by dogeared** [comment]
@mraible @mrioan - I don't think this is an issue directly related to these example updates. The failures you outlined above are present in the `364-spa-support-merged` branch.

We should fix that branch or put in an issue to address it after merge.

If that's the only issue at this point with this branch, I will merge it. 

Let me know if that's ok @mraible

**Comment by mrioan** [comment]
@mraible #661 has not yet been pushed. I am finishing it now so I we can have it for the UAT
